components/amount-buttons.jsx

```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
       this.props.onRadioChange();
     }
   },
+  onInputChange: function(e) {
+    var inputValue = e.currentTarget.value;
+    var amount = "";
+
+    if (/^(\d)*[\,]?(\d){0,2}$/.test(inputValue)) {
```

Comment:
Actually I think this test is not enough. We need to consider about locales that uses comma as a separator like in the US and Canada e.g. 2,000.00 How are we going to deal with this?